liam2/data.py

```diff
@@ -736,14 +739,16 @@ def index_tables(globals_def, entities, fpath):
                             '(but some are declared in the simulation file)')
 
         globals_data = load_path_globals(globals_def)
-
         globals_node = getattr(input_root, 'globals', None)
         for name, global_def in globals_def.iteritems():
             # already loaded from another source (path)
             if name in globals_data:
                 continue
 
             if name not in globals_node:
+                if isinstance(global_def, numbers.Number) or isinstance(global_def, bool):

```

Comment:
wouldn't `if np.isscalar(global_def)` be more robust? I would move the check above, so that None of the code below is executed (it seems useless in this case). In fact, I would just add these lines just below `globals_data = load_path_globals(globals_def)`:
```python
globals_data.update({k: v for k, v in globals_def.iteritems()
                     if np.isscalar(v)})
```
**EDIT**: In fact, I think this code is completely useless now that the constants are handled before the loop (currently in load_path_globals, but if you handle them in `handle_constant_globals`, that would be the same.